extensions/sdk/src/Rally.ts

```diff
@@ -99,9 +99,49 @@ export class Rally {
       connectFirestoreEmulator(this._db, 'localhost', 8080);
     }
 
+    this.storeAttributionCodes()
+      .catch((err) => console.error("Could not store attribution codes:", err));
+
     onAuthStateChanged(this._auth, user => this.authStateChangedCallback(user));
   }
 
+  /**
+   * Attempt to fetch the attribution codes from the store page URL for this extension.
+   */
+  private async storeAttributionCodes() {
+    const attribution = await this.getAttributionCodes();
+    if (!(Object.keys(attribution).length === 0 && attribution.constructor === Object)) {
+      console.debug("Attribution codes already stored");
+      return;
+    }
+
+    // Study IDs are in camelCase, store IDs are in hyphen-case.
+    const camelToHyphenCase = str => str.replace(/[A-Z]/g, letter => `-${letter.toLowerCase()}`);
+    const storeId = camelToHyphenCase(this._options.studyId);
+
+    let storeUrl;
+    if ("getBrowserInfo" in browser.runtime) {
+      const browserInfo = await browser.runtime.getBrowserInfo();
+      if (browserInfo.name === "firefox") {
+        storeUrl = `https://addons.mozilla.org/en-US/firefox/addon/${storeId}/*`
+      }
```

Comment:
Should the below else match up with this if statement? Or can we do this: refactor const strings and then
```
let storeUrl = ChromeUrl; // Default

const browserInfo = brower.runtime && browser.runtime.getBrowserInfo && await browser.runtime.getBrowserInfo();

if (browserInfo && browserInfo.name === "firefox") {
   storeUrl = FireFoxUrl;
}
```

We can likely extract a method for this via refactoring.

Reply:
Sure, we did something similar over in web-science recently, but I'm not sure happy with it.